include/zephyr/net/net_ip.h

```diff
@@ -600,7 +600,7 @@ struct net_tcp_hdr {
 	uint8_t wnd[2];
 	uint16_t chksum;
 	uint8_t urg[2];
-	uint8_t optdata[0];
+	uint8_t optdata[];
```

Comment:
Not sure I understand your criteria for when you replace `[0]` with `[]` and when you replace it with `FLEXIBLE_ARRAY_DECLARE`.

For example here you replace with `[]`, but in `log_multidomain_source_name` you use `FLEXIBLE_ARRAY_DECLARE`. 

Can you elaborate on this?

Reply:
From what I can tell, `FLEXIBLE_ARRAY_DECLARE` is only required if the flexible array member is the only field in a union or a struct. Was the `log_multidomain_source_name` and others an oversight?

Reply:
The macro is safer and makes easier if we need to add anything and since I had to change `log_multidomain_log_msg` I kept the other changes in the file. 

But it can be just `[]`.

Reply:
Right, but the macro has an issue with empty struct in C++, right? So not ideal in header files that support C++